core/primitives/src/block.rs

```diff
@@ -56,6 +56,7 @@ pub struct BlockV1 {
 pub struct BlockV2 {
     pub header: BlockHeader,
     pub chunks: Vec<ShardChunkHeader>,
+    #[deprecated]
     pub challenges: Challenges,
```

Comment:
Is this field still used somewhere or can we rename to `_challenges`?

Reply:
it is used in a couple of places when we have to construct `BlockV2`.  

Renaming the field would cause a protocol schema change.  Is that not a problem?  I suppose as long as the [de]serialisation is not changed.  

Do we have tests that will catch any potential issues a renaming would cause?

Reply:
i'll merge the PR and we can do the renaming a follow up PR as needed.